### DISCUSSION TITLE
wrap prompt line

### DISCUSSION THREAD

**Discussion by rodhash** [junegunn/fzf#3653] [Q&A] [answered]
Any way we can enable this? In the docs I found a "nowrap" only for the preview window but I'm looking for the prompt line.

In normal situations I never needed this but I found myself doing a long filter on a file.. at the end I wanted to see the whole prompt line but the normal behavior is to truncate the prompt line, right?

Ideas?

Thanks

**Comment by LangLangBart** [comment, marked as answer]
> wrap prompt line
> Any way we can enable this?

I believe this feature does not currently exist.

> Ideas?

One alternative solution could be to assign a hotkey that alters the preview to display the '_prompt_'
in the preview.

By '_prompt_', I assume you're referring to the '_query input_', correct?

<img src="https://github.com/junegunn/fzf/assets/92653266/ad955faf-446c-4c6d-8a48-7de6eb0b46e1"
width="400">

```bash
# Generate a random string
long_query=$(LC_ALL=C tr -dc "[:alnum:]" </dev/urandom | head -c 400)
# '{q}' serves as a placeholder for the current query -  see 'man fzf' for further details
: | fzf --query=$long_query --bind 'ctrl-t:change-preview:echo {q}' --preview-window 'wrap'
```
Upon pressing <kbd>⌃ Control</kbd> + <kbd>T</kbd>, the preview will show the current '_query input_'.

<img src="https://github.com/junegunn/fzf/assets/92653266/be7f2d00-cbe4-488c-aa7f-1fa893dafc69"
width="400">

**Reply by rodhash** [reply]
very nice, it works!

now I'm trying to figure out how to switch back to the previous preview

**Reply by rodhash** [reply]
oh nevermind got it, just had to create another binding to change-preview back.. awesome solution

**Reply by rodhash** [reply]
thank you sir
